src/nvim/ex_cmds.c

```diff
@@ -5935,3 +6036,191 @@ void set_context_in_sign_cmd(expand_T *xp, char_u *arg)
+  win_T *oldwin = curwin;
+
+  if (split) {
+    // don't use a new tab page
+    cmdmod.tab = 0;
+
+    // Create a window for the command-line buffer.
+    if (win_split((int)p_cwh, WSP_BOT) == FAIL) {
+      beep_flush();
+      return;
+    }
+    cmdwin_type = get_cmdline_type();
+
+    // Create the command-line buffer empty.
+    (void)do_ecmd(0, NULL, NULL, NULL, ECMD_ONE, ECMD_HIDE, NULL);
+    (void)setfname(curbuf, (char_u *) "[inc_sub]", NULL, true);
+    set_option_value((char_u *) "bt", 0L, (char_u *) "incsub", OPT_LOCAL);
+    set_option_value((char_u *) "swf", 0L, NULL, OPT_LOCAL);
+    curbuf->b_p_ma = false;  // Not Modifiable
+    curwin->w_p_fen = false;
+    curwin->w_p_rl = cmdmsg_rl;
+    cmdmsg_rl = false;
+    RESET_BINDING(curwin);
+
+    // Showing the prompt may have set need_wait_return, reset it.
+    need_wait_return = false;
+
+    // Reset 'textwidth' after setting 'filetype'
+    // (the Vim filetype plugin sets 'textwidth' to 78).
+    curbuf->b_p_tw = 0;
+
+    // Save the buffer used in the split
+    livebuf = curbuf;
+
+    // Initialize line and highlight variables
+    int line = 0;
+    int src_id_highlight = 0;
+    long sub_size = STRLEN(sub);
+    long pat_size = STRLEN(pat);
+
+    // Get the width of the column which display the number of the line
+    linenr_T highest_num_line = kv_last(*lmatch).lnum;
+
+    // computing the length of the column that will display line number
+    int col_width = log10(highest_num_line) + 1 + 3;
+
+    // will be allocated in the loop
+    char *str = NULL;
+
+    size_t old_line_size = 0;
+    size_t line_size;
+
+    // Append the lines to our buffer
+    for (size_t i = 0; i < (*lmatch).size; i++) {
+      MatchedLine mat = (*lmatch).items[i];
+      line_size = STRLEN(mat.line) + col_width + 1;
+
+      // Reallocation if str not long enough
+      if (line_size > old_line_size) {
+        str = xrealloc(str, line_size * sizeof(char));
+        old_line_size = line_size;
+      }
+
+      // put ' [ lnum]line' into str and append it to the incsubstitute buffer
+      snprintf(str, line_size, " [%*ld]%s", col_width - 3, mat.lnum, mat.line);
+      ml_append(line++, (char_u *)str, (colnr_T)line_size, false);
+
+      // highlight the replaced part
+      if (sub_size > 0) {
+        int hlgroup_ls = syn_check_group((char_u *)"IncSubstitute", 13);
+
+        for (size_t j = 0; j < mat.start_col.size; j++) {
+          src_id_highlight =
+          bufhl_add_hl(curbuf,
+                       src_id_highlight,
+                       hlgroup_ls,  // id of our highlight
+                       line,
+                       mat.start_col.items[j] + col_width
+                       + j * (sub_size - pat_size) + 1,
+                       mat.start_col.items[j] + col_width
+                       + j * (sub_size - pat_size) + sub_size);
+        }
+      }
+    }
+    xfree(str);
+  }
+
+  redraw_later(SOME_VALID);
+
+  // Restore the old window
```

Comment:
maybe this also should be in `if(split)` just to make it clear this restores the state from window splitting.